### PR TITLE
Update build.sh

### PR DESCRIPTION
**Pull request by ericLemanissier**
this should separate the various issues more cleanly

### CODE REVIEW

File: build.sh [hunk 1]
```diff
@@ -4,9 +4,10 @@ set -euxo pipefail
 
 SCRIPT_DIR="$( cd "$( dirname "${BASH_SOURCE[0]}" )" >/dev/null 2>&1 && pwd )"
 
-conan export "${SCRIPT_DIR}/lib_a"
-conan export "${SCRIPT_DIR}/lib_b"
-conan export "${SCRIPT_DIR}/lib_c"
+conan create "${SCRIPT_DIR}/lib_c" 
+conan create "${SCRIPT_DIR}/lib_b" 
+conan create "${SCRIPT_DIR}/lib_a"
```
Comment (blackliner) [inline build.sh]:
```suggestion
conan create "${SCRIPT_DIR}/lib_c"
conan create "${SCRIPT_DIR}/lib_b"
conan create "${SCRIPT_DIR}/lib_a"
```

File: build.sh [hunk 2]
```diff
@@ -4,9 +4,10 @@ set -euxo pipefail
...
-conan export "${SCRIPT_DIR}/lib_c"
+conan create "${SCRIPT_DIR}/lib_c" 
+conan create "${SCRIPT_DIR}/lib_b" 
+conan create "${SCRIPT_DIR}/lib_a" 
 
-conan create "${SCRIPT_DIR}/lib_a" -b missing
-conan create "${SCRIPT_DIR}/lib_a" -b missing -o *:shared=True
+conan create "${SCRIPT_DIR}/lib_c"-o *:shared=True
+conan create "${SCRIPT_DIR}/lib_b"-o *:shared=True
+conan create "${SCRIPT_DIR}/lib_a"-o *:shared=True
```
Comment (blackliner) [inline build.sh]:
```suggestion
conan create "${SCRIPT_DIR}/lib_c" -o *:shared=True
conan create "${SCRIPT_DIR}/lib_b" -o *:shared=True
conan create "${SCRIPT_DIR}/lib_a" -o *:shared=True
```

File: lib_a/conanfile.py [hunk 3]
```diff
@@ -44,7 +44,7 @@ def _configure_cmake(self):
 
         self._cmake.definitions["BUILD_TESTING"] = self.options.with_tests
 
-        self._cmake.configure(source_folder="src")
+        self._cmake.configure(source_folder=self.source_folder)
```
Comment (blackliner) [inline lib_a/conanfile.py]:
```suggestion
        self._cmake.configure()
```
 I think this is the default setting, right?

Comment (ericLemanissier) [reply lib_a/conanfile.py]: probably

**Comment by blackliner** [comment]
one thing that wont work with this layout is to run `cmake.test()` right after the build step. To fix this, one would need to move the `include(CTest)` one level up :-/ 

https://cmake.org/cmake/help/latest/module/CTest.html
```
Include this module in the top CMakeLists.txt file of a project to enable testing with CTest and dashboard submissions to CDash:
```

**Comment by ericLemanissier** [comment]
maybe it can be included at both levels ?

**Pull request closed without merging**